### PR TITLE
Move regression tests to evaluation/

### PR DESCRIPTION
**Pull request by zch-cc**
This pr is to address https://github.com/OpenDevin/OpenDevin/issues/76
Changes include:
1. move regression folder to evaluation folder
2. change directory structure to /case/ouputs/agents/worspace or /case/output/agents/logs.txt
3. change run.sh to iterate every agent as well if defined in the case/output/agents folder

### CODE REVIEW

File: evaluation/regression/run.sh
Comment (rbren) [inline evaluation/regression/run.sh:28]: we'll need to pass the agent class into this command, as `--agent-cls`. You might need to hard-code the map from dir to cls

See: https://github.com/OpenDevin/OpenDevin/blob/main/opendevin/main.py

**Review by rbren — APPROVED** [review]

**Review by xingyaoww — CHANGES_REQUESTED** [review]

**Comment by rbren** [comment]
Agree, now that #93 is in, we should use `python opendevin/main.py` instead of `docker run`

File: evaluation/regression/run.sh
Comment (zch-cc) [reply evaluation/regression/run.sh]: I delete this command since we are using python3 directly now

**Review by rbren — APPROVED** [review]
This LGTM! Once it's merged I can run for the codeact agent using OpenAI and see what it does.

**Pull request merged**
